File: python/strings/reverse_string.py
s = input("Enter a string: ")

# Convert to list and reverse using two pointers
arr = list(s)
i, j = 0, len(arr) - 1
while i < j:
    arr[i], arr[j] = arr[j], arr[i]
    i += 1
    j -= 1

reversed_str = ''.join(arr)
print("Reversed string:", reversed_str)

File: java/strings/reverse_string.java
import java.util.Scanner;

public class ReverseString {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        System.out.print("Enter a string: ");
        String str = sc.nextLine();

        // Convert string to char array
        char[] arr = str.toCharArray();

        // Reverse using two pointers
        int i = 0, j = arr.length - 1;
        while (i < j) {
            char temp = arr[i];
            arr[i] = arr[j];
            arr[j] = temp;
            i++;
            j--;
        }

        // Convert back to string
        String reversed = new String(arr);
        System.out.println("Reversed string: " + reversed);
    }
}
